code/game/objects/items/melee/misc.dm

```diff
@@ -540,6 +554,14 @@
 			if (wait_desc)
 				to_chat(user, wait_desc)
 
+/obj/item/melee/classic_baton/contractor_baton/pickup(mob/user)
+	. = ..()
+	if(!owner_data)
+		var/datum/antagonist/traitor/traitor_data = user.mind.has_antag_datum(/datum/antagonist/traitor)
```

Comment:
Again, just make it a check, this does not need a var.

Reply:
yes it does